game-core/src/main/java/games/strategy/engine/lobby/server/User.java

```diff
@@ -3,60 +3,27 @@
 import static com.google.common.base.Preconditions.checkNotNull;
 
 import java.net.InetAddress;
-import java.util.Objects;
 
 import javax.annotation.concurrent.Immutable;
 
+import lombok.Builder;
+import lombok.NonNull;
+import lombok.Value;
+
 /**
  * Information about a lobby user.
  */
 @Immutable
+@Builder
+@Value
 public final class User {
+  @NonNull
```

Comment:
The Lombok Javadocs for this annotation seem to suggest that _any_ annotation with this name (case-insensitive) will have the same effect:

> Note that any annotation named `NonNull` with any casing and any package will result in nullchecks produced for generated methods (and the annotation will be copied to the getter return type and any parameters of generated methods), but only this annotation, if present on a parameter, will result in a null check inserted into your otherwise handwritten method. WARNING: If the java community ever does decide on supporting a single `@NonNull` annotation (for example via JSR305), then this annotation will be deleted from the lombok package. If the need to update an import statement scares you, you should use your own annotation named `@NonNull` instead of this one.

Should we just use the JSR 305 `@Nonnull` for consistency?  If we ever do enable null analysis, Eclipse won't support multiple non-`null` annotations.

Reply:
I'm not sure if it was always the case, but at one point, maybe java 1.7, @Nonnull did not actually change the code generated by lombok So seems like we can stick to @Nonnull happily enough  : )